### PR TITLE
Change abstract method semantics.

### PR DESCRIPTION
**Pull request by floitsch**
If a superclass has the method implemented then a method isn't actually abstract.

### CODE REVIEW

File: tests/negative/abstract6-test.toit [hunk 1]
```diff
@@ -11,5 +11,13 @@ abstract class B extends A:
 
 class C extends B:
 
+mixin M1:
+  foo:
+
+abstract mixin M2 extends M1:
+  abstract foo
+
+
```
Comment (kasperl) [inline tests/negative/abstract6-test.toit:20]: Drop one newline.

File: tests/abstract5-test.toit [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+abstract class D:
+  abstract foo x=null
+  foo: return 1
+
+class E extends D:
+  foo x: return 2
+
+main:
+  c := C
+  expect_equals 1 c.foo
```
Comment (kasperl) [inline tests/abstract5-test.toit:25]:
```suggestion
  expect-equals 1 c.foo
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/abstract5-test.toit (hunk 2 above)
Comment (floitsch) [reply tests/abstract5-test.toit:25]: I will migrate all code in a follow-up PR.

File: tests/negative/abstract6-test.toit (hunk 1 above)
Comment (floitsch) [reply tests/negative/abstract6-test.toit:20]: done.

**Pull request merged**